src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -90,19 +89,19 @@ public function getEnv($prefix, $name, \Closure $getEnv)
         }
 
         if ('bool' === $prefix) {
-            return (bool) self::phpize($env);
+            return (bool) (filter_var($env, FILTER_VALIDATE_BOOLEAN) ?: filter_var($env, FILTER_VALIDATE_INT) ?: filter_var($env, FILTER_VALIDATE_FLOAT));
```

Comment:
why allow ints or floats to be casted as bool, and not limit to bools only ?

Reply:
Because that's supported+tested today, and we shouldn't break BC.